client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/rates-step/list.js

```diff
@@ -29,43 +29,89 @@ const renderRateNotice = translate => {
 	);
 };
 
-export const ShippingRates = ( {
-	id,
-	selectedRates, // Store owner selected rates, not customer
-	availableRates,
-	selectedPackages,
-	allPackages,
-	updateRate,
-	errors,
-	shouldShowRateNotice,
-	translate,
-} ) => {
-	const packageNames = getPackageDescriptions( selectedPackages, allPackages, true );
-	const hasSinglePackage = 1 === Object.keys( selectedPackages ).length;
-
-	const renderSinglePackage = ( pckg, pckgId ) => {
-		if ( ! ( pckgId in availableRates ) ) {
+const MAX_VISIBLE_RATE = 5; // Maximum number of rates to display in the shipping label modal
+
+class ShippingRates extends Component {
+	render() {
+		const packageNames = getPackageDescriptions( this.props.selectedPackages, this.props.allPackages, true );
+		const hasSinglePackage = 1 === Object.keys( this.props.selectedPackages ).length;
+
+		const shippingRates = Object.keys(this.props.selectedPackages).map((pckgId) => {
+			return <PackageShippingRates
+				key={pckgId}
+				id={this.props.id}
+				updateRate={this.props.updateRate}
+				translate={this.props.translate}
+				pckgId={pckgId}
+				selectedRate={this.props.selectedRates[ pckgId ] || ''}  // Store owner selected rates, not customer
+				availableRates={this.props.availableRates[ pckgId ]}
+				packageErrors = {this.props.errors[ pckgId ] || []}
+				packageNames = {packageNames[ pckgId ]}
+				hasSinglePackage = { hasSinglePackage }
+			/>
+		});
+		return <div>
+			{ this.props.shouldShowRateNotice && renderRateNotice( this.props.translate ) }
+			{ shippingRates }
+		</div>
+	}
+}
+
+class PackageShippingRates extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			packageRates: JSON.parse(JSON.stringify(this.props.availableRates.default.rates)),
```

Comment:
What is the purpose of this line?

Reply:
This is to handle object references for price summary. I added a comment in the commit here: https://github.com/Automattic/woocommerce-services/pull/1781/commits/5847be0f4d3413f34a9db50edbf854abbb694d3e